# Question 14
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
full_time = normal(40, 4.5)
part_time = normal(20, 6.9)
FG32 = full_time.integrate(32, 2000)
FTworkers = FG32 * 100
PG32 = part_time.integrate(32, 2000)
PTworkers = PG32 * 200
workers = FTworkers + PTworkers
workers = round(workers, 0)
print("workers > 32 hours = " + str(workers))

workers > 32 hours = 104.0


In [3]:
z = normal(0, 1)
loc = z.locate(0.06)
T = (4.5 * loc) + 40
print("T1 = " + str(T))

T1 = 33.003579999731365


In [4]:
loc = z.locate(0.97)
T_2 = (6.9 * loc) + 20
print("T2 = " + str(T_2))

T2 = 32.97758899962655


In [5]:
a1 = 1 - 0.015
loc_1 = z.locate(a1)
a2 = 1 - 0.166
loc_2 = z.locate(a2)
print(loc_1)
print(loc_2)

2.1701099999477713
0.9701099999402417


$$1434 - 2.17\sigma = \mu$$

$$1194 - 0.97\sigma = \mu$$

$$1434 - 2.17\sigma = 1194 - 0.97\sigma$$

$$1.2\sigma = 240$$

$$\sigma = 200$$

$$\mu = (1434 - (2.17 * 200))$$

$$\mu = 1000$$